packages/cli/src/commands/replay/replay.command.ts

```diff
@@ -312,23 +284,134 @@ export const replayCommandHandler: (
   return replay;
 };
 
-const handler: (options: ReplayCommandHandlerOptions) => Promise<void> = async (
-  options
-) => {
-  await replayCommandHandler({ ...options, exitOnMismatch: true });
+const serveOrGetAppUrl = async (
+  client: AxiosInstance,
+  replayTarget: ReplayTarget
+): Promise<{ appUrl?: string; closeServer?: () => void }> => {
+  if (replayTarget.type === "snapshotted-assets") {
+    const server = await serveAssetsFromSimulation(
+      client,
+      replayTarget.simulationIdForAssets
+    );
+    return {
+      appUrl: server.url,
+      closeServer: server.closeServer,
+    };
+  }
+  if (replayTarget.type === "url") {
+    return { appUrl: replayTarget.appUrl };
+  }
+  if (replayTarget.type === "original-recorded-url") {
+    return {};
+  }
+  return assertNever(replayTarget);
+};
+
+const assertNever = (value: never): never => {
```

Comment:
I don't think we use this pattern anywhere yet, but this is a common ts trick used to get the compiler to tell you if you've forgotten to handle a discriminated type union case anywhere (useful in the case we added in future e.g. a fourth type of replay target; the compiler would force us to update all the right places) .e.g https://stackoverflow.com/questions/39419170/how-do-i-check-that-a-switch-block-is-exhaustive-in-typescript

Side note: with strictNullChecks enabled and a function that has a required return type you can just omit the last return statement, and get the same effect. However we have a compiler setting or eslint rule that forces functions that return a value to have a final return statemen -- but assertNever works.

Reply:
If we find ourselves using this a lot we can put it in a common package.